scene/gui/text_edit.cpp

```diff
@@ -251,13 +251,18 @@ void TextEdit::Text::clear() {
 	insert(0, "");
 }
 
-int TextEdit::Text::get_max_width() const {
+int TextEdit::Text::get_max_width(bool p_exclude_hidden) const {
 	//quite some work.. but should be fast enough.
 
 	int max = 0;
-
-	for (int i = 0; i < text.size(); i++)
-		max = MAX(max, get_line_width(i));
+	if (p_exclude_hidden) {
```

Comment:
Just to mention, it would be ok to move the test for p_exclude_hidden in the loop. This if usually optimized by the compiler. See "Unswitching" here: https://en.wikipedia.org/wiki/Loop_optimization.
That would allow you to reduce the number of code line (which usually helps readability and thus maintenance) :)